lib/transaction.js

```diff
@@ -17,27 +18,45 @@ var Utils = require('./utils')
  * @param {String} options.isolationLevel=true Sets the isolation level of the transaction.
  * @param {String} options.deferrable Sets the constraints to be deferred or immediately checked.
  */
-var Transaction = module.exports = function(sequelize, options) {
+var Transaction = module.exports = function (sequelize, options) {
   this.sequelize = sequelize;
   this.savepoints = [];
   this.options = Utils._.extend({
     autocommit: true,
     type: sequelize.options.transactionType,
     isolationLevel: sequelize.options.isolationLevel
   }, options || {});
+  if (sequelize.options.dialect !== 'mssql') {
 
-  this.parent = this.options.transaction;
-  this.id = this.parent ? this.parent.id : uuid.v4();
+    this.parent = this.options.transaction;
+    this.id = this.parent ? this.parent.id : uuid.v4();
```

Comment:
I will clear those. I have new implementation . I will push that soon. I
need to test transaction with other provider  like MYSQL, SQLITE etc... for
MS SQL my code is working.

On Wed, Jun 22, 2016 at 12:56 PM, Jan Aagaard Meier <
notifications@github.com> wrote:

> In lib/transaction.js
> <https://github.com/sequelize/sequelize/pull/5959#discussion_r68005505>:
>
> >
> > -  this.parent = this.options.transaction;
> > -  this.id = this.parent ? this.parent.id : uuid.v4();
> > +    this.parent = this.options.transaction;
> > +    this.id = this.parent ? this.parent.id : uuid.v4();
>
> Yep, a lot of unnecessary duplication going on here ..
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sequelize/sequelize/pull/5959/files/9f6aaf02b4eb122f788a9fed45bf96c5409f6066#r68005505>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIXpX8sY2O7EYndABVDPoweQFNIwfyLYks5qOOOLgaJpZM4ImY7L>
> .
>



-- 
Contact# : 9426442084
Skype : govinda.alwani
Thank you
with best regards.